openapi/openapi/src/main/scala/endpoints4s/openapi/JsonSchemas.scala

```diff
@@ -58,14 +58,74 @@ trait JsonSchemas extends algebra.JsonSchemas with TuplesSchemas {
 
   object DocumentedJsonSchema {
 
-    case class DocumentedRecord(
-        fields: List[Field],
-        additionalProperties: Option[DocumentedJsonSchema] = None,
-        name: Option[String] = None,
-        description: Option[String] = None,
-        example: Option[ujson.Value] = None,
-        title: Option[String] = None
-    ) extends DocumentedJsonSchema
+    sealed abstract class DocumentedRecord extends DocumentedJsonSchema {
+      def fields: List[Field]
+      def additionalProperties: Option[DocumentedJsonSchema]
+      def name: Option[String]
+
+      def withName(name: String): DocumentedRecord
+      def withExample(example: => ujson.Value): DocumentedRecord
+      def withTitle(title: String): DocumentedRecord
+      def withDescription(description: String): DocumentedRecord
+    }
+    object DocumentedRecord {
+
+      case class Immediate(
+          fields: List[Field],
+          additionalProperties: Option[DocumentedJsonSchema] = None,
+          name: Option[String] = None,
+          description: Option[String] = None,
+          example: Option[ujson.Value] = None,
+          title: Option[String] = None
+      ) extends DocumentedRecord {
+        def withName(name: String): DocumentedRecord =
+          copy(name = Some(name))
+        def withExample(example: => ujson.Value): DocumentedRecord =
+          copy(example = Some(example))
+        def withTitle(title: String): DocumentedRecord =
+          copy(title = Some(title))
+        def withDescription(description: String): DocumentedRecord =
+          copy(description = Some(description))
+      }
+
+      class Lazy(n: String, docs: => DocumentedRecord) extends DocumentedRecord {
+        lazy val evaluatedDocs = docs
+        def fields: List[Field] = evaluatedDocs.fields
+        def additionalProperties: Option[DocumentedJsonSchema] =
+          evaluatedDocs.additionalProperties
+        def name: Option[String] = Some(n)
+        def description: Option[String] = evaluatedDocs.description
+        def example: Option[ujson.Value] = evaluatedDocs.example
+        def title: Option[String] = evaluatedDocs.title
+
+        def withName(n: String): DocumentedRecord =
+          new Lazy(n, this)
```

Comment:
Should it be the following, just to avoid one more level of indirection?

```suggestion
          new Lazy(n, docs)
```

Reply:
Indeed. For `withName` we can avoid that indirection.